go/vt/topo/zk2topo/server.go

```diff
@@ -100,14 +107,32 @@ func (zs *Server) connForCell(ctx context.Context, cell string) (Conn, string, e
 	if err := proto.Unmarshal(data, ci); err != nil {
 		return nil, "", fmt.Errorf("cannot Unmarshal CellInfo for cell %v: %v", cell, err)
 	}
+	serverAddress := cellOverride(cell)
+	if serverAddress == "" {
+		serverAddress = ci.ServerAddress
+	}
 	ins = &instance{
 		root: ci.Root,
-		conn: Connect(ci.ServerAddress),
+		conn: Connect(serverAddress),
 	}
 	zs.instances[cell] = ins
 	return ins.conn, ins.root, nil
 }
 
+func cellOverride(cell string) string {
+	for _, o := range strings.Split(*cellOverrides, ",") {
+		if strings.HasPrefix(cell, o) {
```

Comment:
The order of these arguments is incorrect :(. Should be `strings.HasPrefix(o, cell)`.